lib/features/notes/presentation/widgets/daily_reminders.dart

```diff
@@ -0,0 +1,154 @@
+import 'package:dairy_app/app/themes/theme_extensions/note_create_page_theme_extensions.dart';
+import 'package:dairy_app/app/themes/theme_extensions/settings_page_theme_extensions.dart';
+import 'package:dairy_app/core/dependency_injection/injection_container.dart';
+import 'package:dairy_app/core/utils/utils.dart';
+import 'package:dairy_app/features/auth/core/constants.dart';
+import 'package:dairy_app/features/auth/data/models/user_config_model.dart';
+import 'package:dairy_app/features/auth/presentation/bloc/user_config/user_config_cubit.dart';
+import 'package:dairy_app/features/notes/domain/repositories/notifications_repository.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:simple_accordion/widgets/AccordionHeaderItem.dart';
+import 'package:simple_accordion/widgets/AccordionItem.dart';
+import 'package:simple_accordion/widgets/AccordionWidget.dart';
+
+// ignore: must_be_immutable
+class DailyReminders extends StatelessWidget {
+  late INotificationsRepository notificationsRepository;
+
+  DailyReminders({Key? key}) : super(key: key) {
+    notificationsRepository = sl<INotificationsRepository>();
```

Comment:
Please move this into the build method and make this into a const constructor.

Reply:
the tradeoff here is 

If we initialize in build method and make the widget const
1. It can prevent unnecessary rebuilds causing from settings page

If we initialize it in constructor and keep remove const 
1. It has to fetch notificationRepository from `get_it` everytime build method runs which might have some overhead.
2. notificationRepository can be safely accessed from any function other than build

Not sure which one is better here, I went with initializing in constructor mostly because of 2nd point

Reply:
I guess accessing directly from get_it is fast, we can access it like `sl<INotificationsRepository>()` every time if we have to access it outside the build method. Will add this change.
<img width="1083" alt="image" src="https://github.com/SankethBK/diaryvault/assets/51091231/bf1d51a2-0e40-4830-ba05-8bdc3f8dc9a0">